### PR TITLE
Split app into modules

### PR DESCRIPTION
**Pull request by mdrlzy**
### Modules structure
```
ARK-Rate
├── app
├── core
│   ├── data
│   ├── db
│   ├── di
│   ├── domain
│   └── presentation
├── feature
│   ├── pairalert
│   ├── portfolio
│   ├── quick
│   ├── quickwidget
│   ├── search
│   └── settings
├── fiaticons
└── cryptoicons

```

### Resources
All resources are located in `:core:presentation`. Keeping resources in feature modules and ensuring that they are not duplicated will be a mess. We can easily access them via type aliases.
```
CoreR.kt

typealias CoreRString = R.string
typealias CoreRDrawable = R.drawable
```
### Navigation
Feature modules generate compose-destination classes. We collect them in app module and create a single graph manually.
`compose-destinations` library works in destinations mode
```
ksp {
    arg("compose-destinations.mode", "destinations")
}
```
### DI
We have `CoreComponent` in `:core:di` that contains common dependencies with `Singleton` scope. Feature modules create own components that depend on `CoreComponent` with own scopes. App holds `CoreComponent` and provides it. 

Feature component are stored in separate object called `FeatureComponentHolder` inside the module. This is the easiest way to reuse component between compose screens in module.

### Notes

- `:core:di` and `:core:db` needed to not expose `:core:data` module and implementation so feature modules only know about `:core:domain` and interfaces
- Build config replaced with `BuildConfigFields` and `BuildConfigFieldsProvider`. `BuildConfigFieldsProvider` created in `:core:data` module and initialized in app module. So all modules can use the build config of app module.

### CODE REVIEW

**Review by shubertm — CHANGES_REQUESTED** [review]

File: core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class TimestampRepoImpl @Inject constructor(private val dao: TimestampDao) : TimestampRepo {
+    override suspend fun rememberTimestamp(type: TimestampType) =
+        dao.insert(
+            RoomFetchTimestamp(
+                type.name,
+                OffsetDateTime.now(),
+            ),
+        )
+
+    override suspend fun getTimestamp(type: TimestampType) =
```
[the remark this answers is not in the document]
Comment (mdrlzy) [reply core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt:23]: Can you give an argument? Other modules use `TimestampRepo` where return types are specified and don't know about `TimestampRepoImpl`. Also this is allowed by ktlint.

File: core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import dev.arkbuilders.rate.core.domain.repo.TimestampRepo
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import java.time.OffsetDateTime
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TimestampRepoImpl @Inject constructor(private val dao: TimestampDao) : TimestampRepo {
+    override suspend fun rememberTimestamp(type: TimestampType) =
```
[the remark this answers is not in the document]
Comment (mdrlzy) [reply core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt:15]: Same as `getTimestamp`

File: core/presentation/src/main/java/dev/arkbuilders/rate/core/presentation/ui/DropDown.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    title: String,
+    icon: Painter,
+) {
+    Row(
+        modifier =
+            modifier
+                .height(44.dp)
+                .border(
+                    1.dp,
+                    ArkColor.Border,
```
[the remark this answers is not in the document]
Comment (mdrlzy) [reply core/presentation/src/main/java/dev/arkbuilders/rate/core/presentation/ui/DropDown.kt:35]: I don't see much difference, anyway we will refactor this to use colors from `Theme` to support dark theme

File: core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (shubertm) [reply core/data/src/main/java/dev/arkbuilders/rate/core/data/repo/TimestampRepoImpl.kt:23]: Just for convenience, look `rememberTimestamp` returns `Unit` and `getTimestamp` looks like it returns `Unit` as well

File: core/presentation/src/main/java/dev/arkbuilders/rate/core/presentation/ui/DropDown.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (shubertm) [reply core/presentation/src/main/java/dev/arkbuilders/rate/core/presentation/ui/DropDown.kt:35]: Alright, cool

**Comment by shubertm** [comment]
https://github.com/user-attachments/assets/9447f6c7-09aa-43bd-b096-344ee3b635c9

- [ ] Trying to change the "from currency" creates new "to currency"

**Review by shubertm — APPROVED** [review]

**Pull request merged**